src/main/java/com/juniorstart/juniorstart/service/UserAdditionalSkillService.java

```diff
@@ -0,0 +1,75 @@
+package com.juniorstart.juniorstart.service;
+
+import com.juniorstart.juniorstart.exception.ResourceNotFoundException;
+import com.juniorstart.juniorstart.model.UserAdditionalSkill;
+import com.juniorstart.juniorstart.model.UserTechnology;
+import com.juniorstart.juniorstart.repository.UserAdditionalSkillRepository;
+import com.juniorstart.juniorstart.repository.UserProfileRepository;
+import com.juniorstart.juniorstart.repository.UserTechnologyRepository;
+import com.juniorstart.juniorstart.security.UserPrincipal;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+
+@Service
+@Slf4j
+public class UserAdditionalSkillService {
+
+    final private UserTechnologyRepository userTechnologyRepository;
+    final private UserProfileRepository userProfileRepository;
+    final private UserAdditionalSkillRepository userAdditionalSkillRepository;
+
+    UserAdditionalSkillService(UserTechnologyRepository userTechnologyRepository, UserProfileRepository userProfileRepository,
+                               UserAdditionalSkillRepository userAdditionalSkillRepository) {
+        this.userTechnologyRepository = userTechnologyRepository;
+        this.userProfileRepository = userProfileRepository;
+        this.userAdditionalSkillRepository = userAdditionalSkillRepository;
+    }
+
+    public ResponseEntity<?> addUserAdditionalSKill(Set<UserAdditionalSkill> userAdditionalSkill, UserTechnology userTechnology , UserPrincipal currentUser) {
```

Comment:
Nie lepiej opakować UserAdditionalSkill w jakiś obiekt?

Reply:
Po konsultacji, z Adrianem, czy jest potrzebny na to jakiś osobny payload ustaliliśmy, że nie ma sensu, przy tak znikomej ilości pól.

Reply:
Jest taka niepisana zasada, że do 3 pól można dawać do konstruktora metody.